packages/datagateway-download/src/downloadApiHooks.ts

```diff
@@ -736,3 +738,35 @@ export const useAdminUpdateDownloadStatus = (): UseMutationResult<
     }
   );
 };
+
+/**
+ * Queries the progress of a {@link Download}.
+ * @param prepareId The prepare ID of the {@link Download}.
+ * @param queryOptions Optional `useQuery` option override.
+ */
+export const useDownloadPercentageComplete = <T = DownloadProgress>({
+  preparedId,
+  ...queryOptions
+}: { preparedId: string } & UseQueryOptions<
+  DownloadProgress,
+  AxiosError,
+  T,
+  string[]
+>): UseQueryResult<T, AxiosError> => {
+  const { idsUrl } = React.useContext(DownloadSettingsContext);
```

Comment:
Just had a thought - each download could have different IDSes and so we can't just get the "general" `idsUrl` from the settings file - we need the one corresponding to the download. We'll probably have to match the `transport` item in a download to the access methods array in the settings to get the right URL.